bento_beacon/utils/handover_utils.py

```diff
@@ -1,68 +1,36 @@
-from flask import current_app, request, url_for
+from flask import current_app
 import requests
 from urllib.parse import urlsplit, urlunsplit
 from .katsu_utils import katsu_network_call
 from .exceptions import APIException
-from .nested_query_utils import auth_header_from_request
+from ..authz.headers import auth_header_from_request
 
-# path elements removed by bento gateway
-# BEACON_PATH_FRAGMENT = "api/beacon"
 
 DRS_TIMEOUT_SECONDS = 10
 
-# may or may not be needed
-# def get_handover_url():
-#     base_url_components = urlsplit(request.url)
-#     handover_scheme = "https"
-#     handover_path = BEACON_PATH_FRAGMENT + url_for("handover.get_handover")
-#     handover_base_url = urlunsplit((
-#         handover_scheme,
-#         base_url_components.netloc,
-#         handover_path,
-#         base_url_components.query,
-#         base_url_components.fragment
-#     ))
-#     return handover_base_url
-
-
-def drs_internal_url_components():
-    return urlsplit(current_app.config["DRS_INTERNAL_URL"])
-
-
-def drs_external_url_components():
-    return urlsplit(current_app.config["DRS_EXTERNAL_URL"])
-
-# TODO: either remove or deduplicate with below
-# def drs_internal_file_link_for_id(id):
-#     internal_url_components = drs_internal_url_components()
-#     path = internal_url_components.path + "/objects/" + id + "/download"
-#     return urlunsplit((
-#         internal_url_components.scheme,
-#         internal_url_components.netloc,
-#         path,
-#         internal_url_components.query,
-#         internal_url_components.fragment
-#     ))
-
-
-def drs_external_file_link_for_id(id):
-    external_url_components = drs_external_url_components()
-    path = external_url_components.path + "/objects/" + id + "/download"
+
+def drs_url_components():
+    return urlsplit(current_app.config["DRS_URL"])
+
+
+def drs_file_link_for_id(id):
+    url_components = drs_url_components()
+    path = url_components.path + "/objects/" + id + "/download"
```

Comment:
this should already be built in the DRS response I think?

Reply:
(via the `access_methods` array)

Reply:
ah, yes, thanks for this, will refactor